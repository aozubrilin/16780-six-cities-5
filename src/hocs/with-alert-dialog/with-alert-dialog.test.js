import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import withAlertDialogConnected, {withAlertDialog} from "./with-alert-dialog";


const MockComponent = () => <div>Mock component</div>;
const MockComponentWrapped = withAlertDialog(MockComponent);
const MockComponentWrappedConnected = withAlertDialogConnected(MockComponent);
const testStore = configureStore();

const noop = () => {};
describe(`withAlertDialog snapshot`, () => {

  describe(`withAlertDialog not connected to store`, () => {

    it(`Not should WithAlertDialog add Alert`, () => {
      const tree = renderer.create((
        <MockComponentWrapped
          closeAlertAction={noop}/>
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`Should WithAlertDialog add Alert`, () => {
      const tree = renderer.create((
        <MockComponentWrapped
          alertMessage={`Message`}
          closeAlertAction={noop}
        />
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`withAlertDialog connected to store`, () => {

    it(`Not should withAlertDialog add Alert`, () => {
      const store = testStore({
        data: {
          errorFetchMessadge: null,
        }
      });

      const tree = renderer.create(
          <Provider store={store}>
            <MockComponentWrappedConnected/>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`Should withAlertDialog add Alert`, () => {

      const store = testStore({
        data: {
          errorFetchMessadge: `Text`,
        }
      });

      const tree = renderer.create(
          <Provider store={store}>
            <MockComponentWrappedConnected/>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});

