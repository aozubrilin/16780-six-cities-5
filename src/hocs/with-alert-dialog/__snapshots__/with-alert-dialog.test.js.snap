// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAlertDialog snapshot withAlertDialog connected to store Not should withAlertDialog add Alert 1`] = `
<div>
  Mock component
</div>
`;

exports[`withAlertDialog snapshot withAlertDialog connected to store Should withAlertDialog add Alert 1`] = `
Array [
  <div
    className="alert"
  >
    <span
      className="closebtn"
      onClick={[Function]}
    >
      ×
    </span>
    Text
  </div>,
  <div>
    Mock component
  </div>,
]
`;

exports[`withAlertDialog snapshot withAlertDialog not connected to store Not should WithAlertDialog add Alert 1`] = `
<div>
  Mock component
</div>
`;

exports[`withAlertDialog snapshot withAlertDialog not connected to store Should WithAlertDialog add Alert 1`] = `
Array [
  <div
    className="alert"
  >
    <span
      className="closebtn"
      onClick={[Function]}
    >
      ×
    </span>
    Message
  </div>,
  <div>
    Mock component
  </div>,
]
`;
